lux-lib/src/rockspec/lua_dependency.rs

```diff
@@ -179,12 +179,121 @@ pub enum DependencyType<T> {
     External(HashMap<String, ExternalDependencySpec>),
 }
 
+impl<T> IntoLua for DependencyType<T>
+where
+    T: IntoLua,
+{
+    fn into_lua(self, lua: &mlua::Lua) -> mlua::Result<mlua::Value> {
+        let table = lua.create_table()?;
+
+        match self {
+            DependencyType::Regular(deps) => {
+                table.set("regular", deps)?;
+            }
+            DependencyType::Build(deps) => {
+                table.set("build", deps)?;
+            }
+            DependencyType::Test(deps) => {
+                table.set("test", deps)?;
+            }
+            DependencyType::External(deps) => {
+                table.set("external", deps)?;
+            }
+        }
```

Comment:
suggestion: Perhaps the table should have a `type` field?
This is quite common in Lua representations of ADTs (similar to JSON tag fields).

Reply:
I've implemented it this way prior in the entire Lua API, but I'm perfectly fine with refactoring that. We should make a tracking issue for such a refactor.

Reply:
#658